core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -298,6 +300,22 @@ public DynamicFilter createDynamicFilter(
                 .map(probeSymbol -> requireNonNull(columnHandles.get(probeSymbol), () -> "Missing probe column for " + probeSymbol))
                 .collect(toImmutableSet());
 
+        Map<DynamicFilterId, OptionalLong> dynamicFilterTimeouts = dynamicFilterDescriptors.stream()
+                .collect(toImmutableMap(
+                        DynamicFilters.Descriptor::getId,
+                        DynamicFilters.Descriptor::getPreferredTimeout,
+                        (timeout1, timeout2) -> {
+                            if (timeout1.isPresent() && timeout2.isPresent()) {
```

Comment:
you could use `io.trino.util.Optionals#combine`

Reply:
It would require similar function which is based on OptionalLong.